plugin/validator_test.go

```diff
@@ -78,6 +78,109 @@ func TestValidator_LoadSchema(t *testing.T) {
+				`(characters outside of the reserved list of RFC 3986 found)"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "/path/200#"))
+		assert.Equal(t,
+			`{"config":{"path":"invalid path: '/path/200#' `+
+				`(characters outside of the reserved list of RFC 3986 found)"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "/foo//bar"))
+		assert.Equal(t, `{"config":{"path":"must not have empty segments"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "/foo/bar//"))
+		assert.Equal(t, `{"config":{"path":"must not have empty segments"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "//foo/bar"))
+		assert.Equal(t, `{"config":{"path":"must not have empty segments"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "[[/users/|foo/profile]]"))
+		assert.Equal(t, `{"config":{"path":"should start with: /"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, "[[/users/(this|foo/profile)]]"))
+		assert.Equal(t, `{"config":{"path":"should start with: /"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%2words`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%2w'"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%0Xwords`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%0X'"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%2Gwords`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%2G'"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%20words%`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%'"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%20words%a`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%a'"}}`, err.Error())
+	})
+	t.Run("errors out on path containing not allowed elements", func(t *testing.T) {
+		err := v.Validate(fmt.Sprintf(plugin, `/some%20words%ax`))
+		assert.Equal(t, `{"config":{"path":"invalid url-encoded value: '%ax'"}}`, err.Error())
+	})
```

Comment:
GitHub won't let me make a suggestion on this big of a code segment I guess. Let's use a struct here for the "badPaths" to make this a little more readable and also check to make sure the error returned is JSON using require.JSONEq. Granted the error should always be JSON, but this will make sure that doesn't change on us.

```
	badPaths := []struct {
		name        string
		path        string
		expectedErr string
	}{
		{
			name: "errors out on empty path",
			path: "",
			expectedErr: `{
				"config": {
					"path": "length must be at least 1"
				}
			}`,
		},
		{
			name: "errors out on path not starting with /",
			path: "path",
			expectedErr: `{
				"config": {
					"path": "should start with: /"
				}
			}`,
		},
		{
			name: "errors out on path not starting with /",
			path: "path/200",
			expectedErr: `{
				"config": {
					"path": "should start with: /"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/path/200?",
			expectedErr: `{
				"config": {
					"path": "invalid path: '/path/200?' (characters outside of the reserved list of RFC 3986 found)"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/path/200#",
			expectedErr: `{
				"config": {
					"path": "invalid path: '/path/200#' (characters outside of the reserved list of RFC 3986 found)"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/foo//bar",
			expectedErr: `{
				"config": {
					"path": "must not have empty segments"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/foo/bar//",
			expectedErr: `{
				"config": {
					"path": "must not have empty segments"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "//foo/bar",
			expectedErr: `{
				"config": {
					"path": "must not have empty segments"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "[[/users/|foo/profile]]",
			expectedErr: `{
				"config": {
					"path": "should start with: /"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "[[/users/(this|foo/profile)]]",
			expectedErr: `{
				"config": {
					"path": "should start with: /"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%2words",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%2w'"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%0Xwords",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%0X'"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%2Gwords",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%2G'"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%20words%",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%'"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%20words%a",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%a'"
				}
			}`,
		},
		{
			name: "errors out on path containing not allowed elements",
			path: "/some%20words%ax",
			expectedErr: `{
				"config": {
					"path": "invalid url-encoded value: '%ax'"
				}
			}`,
		},
	}
	for _, path := range badPaths {
		t.Run(path.name, func(t *testing.T) {
			err := v.Validate(fmt.Sprintf(plugin, path.path))
			assert.NotNil(t, err)
			require.JSONEq(t, path.expectedErr, err.Error())
		})
	}
```

Reply:
I'll do, thx!